### ISSUE TITLE
Flips/crashes in SITL

### ISSUE THREAD

**Issue by julianoes** [PX4/Firmware#14997]
I stumbled upon weird failures in SITL with lockstep at 20x where the vehicle flips during RTL.

I've seen this at least twice now, and it's potentially related to #14835.

https://logs.px4.io/3d?log=4d4c672d-2330-4ab6-9c1d-215c09f4bbd7

**Comment by julianoes** [comment]
In the logfiles I see long drops where it supposedly doesn't log at all. After a first one it doesn't crash but after the second one it does.

**Comment by MaEtUgR** [comment]
In 3D replay it looks really strange. The drone moves in a diagonal very straight line to where it flips on the ground
![image](https://user-images.githubusercontent.com/4668506/83259639-b7ec6a00-a1b8-11ea-812e-f9c0a18ce06c.png)

**Comment by MaEtUgR** [comment]
Like you wrote from the logs it looks like there are two blackouts. The first one is during descending and the second one is when it already hit the ground:
![image](https://user-images.githubusercontent.com/4668506/83260357-e9196a00-a1b9-11ea-8dae-faa60860e377.png)

**Comment by julianoes** [comment]
Right, so the dropouts could just be that the logging doesn't keep up but everything else still runs. Presumably, with lockstep there shouldn't be a case where the simulation continues if there are no more motor outputs, so said differently the CPU being busy should not matter for controls purposes.
